src/lib.rs

```diff
@@ -26,6 +26,10 @@ use core::{fmt, ptr::NonNull};
 pub mod export;
 mod impls;
 mod leb;
+
+#[cfg(armv6m)]
+compile_error! {"defmt cannot be used on Cortex-M0(+) chips, because it requires an ITM peripheral"}
```

Comment:
Did you mean to put this in the `defmt-itm` crate instead? defmt should work on thumbv6m without problems.

Reply:
Oh, yes 😓